### PR TITLE
Refs #32508 -- Raised Type/ValueError instead of using "assert" in django.core.

### PR DESCRIPTION
**Pull request by abbasidaniyal**

### CODE REVIEW

**Comment by felixxm** [comment]
@abbasidaniyal Thanks :+1: Can you also audit other uses of `assert` in `django/core`?:
- `django/core/mail/message.py`: `assert content is None`
- `django/core/mail/message.py`: `assert mimetype is None`
- `django/core/mail/message.py`: `assert content is not None`
- `django/core/mail/message.py`:  `assert content is not None`
- `django/core/mail/message.py`: `assert mimetype is not None`
- `django/core/files/storage.py`: `assert name, "The name argument is not allowed to be empty."`

and add tests for uncovered error messages.

**Comment by abbasidaniyal** [comment]
> @abbasidaniyal Thanks 👍 Can you also audit other uses of `assert` in `django/core`?:
> 
> * `django/core/mail/message.py`: `assert content is None`
> * `django/core/mail/message.py`: `assert mimetype is None`
> * `django/core/mail/message.py`: `assert content is not None`
> * `django/core/mail/message.py`:  `assert content is not None`
> * `django/core/mail/message.py`: `assert mimetype is not None`
> * `django/core/files/storage.py`: `assert name, "The name argument is not allowed to be empty."`
> 
> and add tests for uncovered error messages.

Sure! I was thinking of adding the tests too. Didn't find any previously hence skipped it. 
On it now!

**Comment by abbasidaniyal** [comment]
@felixxm It seems as if `CheckMessage` and it's sub classes (Debug, Info etc) do not have any tests. 
I was planning on creating a new file unders `tests/check_framework` named `test_messages.py`. Let me know if that's alright. 

In the meantime, I'm working on the other instances of assert in django.core.

**Comment by felixxm** [comment]
> @felixxm It seems as if `CheckMessage` and it's sub classes (Debug, Info etc) do not have any tests.

You can add a test to the `check_framework.tests.MessageTests`.

File: django/core/checks/messages.py [hunk 1]
```diff
@@ -9,7 +9,8 @@
 class CheckMessage:
 
     def __init__(self, level, msg, hint=None, obj=None, id=None):
-        assert isinstance(level, int), "The first argument should be level."
+        if not isinstance(level, int):
+            raise TypeError("The first argument should be level.")
```
Comment (abbasidaniyal) [inline django/core/checks/messages.py:13]: We probably should update the error message too.
Something along the lines of level must be of type int?

Comment (felixxm) [reply django/core/checks/messages.py:13]: The current message sounds good to me.

File: tests/mail/tests.py [hunk 2]
```diff
@@ -46,6 +47,11 @@ def assertMessageHasHeaders(self, message, headers):
                         'the following headers: %s' % (headers - msg_headers),)
 
 
+class MIMEFileTest(MIMEBase):
+    def __init__(self):
+        super().__init__("text", "html")
+
+
```
Comment (felixxm) [inline tests/mail/tests.py:54]: A new class is not necessary, you can use `MIMEText`.

**Review by felixxm — COMMENTED** [review]
@abbasidaniyal Thanks :+1: 

I pushed edits, simplified tests, and added small elease notes.

**Pull request merged**